system_tests/common_test.go

```diff
@@ -159,18 +233,74 @@ type NodeBuilder struct {
 	l1StackConfig *node.Config
 	l2StackConfig *node.Config
 	valnodeConfig *valnode.Config
+	l3Config      *NitroConfig
 	L1Info        info
 	L2Info        info
-
-	// L1, L2 Node parameters
-	dataDir       string
-	isSequencer   bool
-	takeOwnership bool
-	withL1        bool
+	L3Info        info
+
+	// L1, L2, L3 Node parameters
+	dataDir                     string
+	isSequencer                 bool
+	takeOwnership               bool
+	withL1                      bool
+	addresses                   *chaininfo.RollupAddresses
+	l3Addresses                 *chaininfo.RollupAddresses
+	initMessage                 *arbostypes.ParsedInitMessage
+	l3InitMessage               *arbostypes.ParsedInitMessage
+	withProdConfirmPeriodBlocks bool
+	useL1StackConfig            bool // don't overwrite the L1 stack config when building
```

Comment:
This field was added so that we can have custom L1 configs for the Espresso tests. Otherwise the custom configs are overwritten by the upstream test harness.